### PR TITLE
Some mkapollo.sh enhancements

### LINKED ISSUE #31: Early boot is black and white
**Describe the bug**
The screenmode in early boot is black and white

**To Reproduce**
1. Start while holding both mousebuttons to get to early boot screen
2. Click on 'Boot without startup sequence'

**Expected behavior**
The screenmode should at least have 4 colors

**Screenshots**
Image of issue

**Architecture**
 - Amiga (including UAE, Vampire cards)

**CPU**
 - m68k

**Version**
Release 5

### PR DESCRIPTION
**Pull request by MBeijer**

### CODE REVIEW

**Comment by ronybeck** [comment]
Can someone explain to me what exactly the goal of this script is?

**Comment by MBeijer** [comment]
Fixes #31

**Comment by MBeijer** [comment]
@ronybeck 
> Can someone explain to me what exactly the goal of this script is?

The goal of the `mkapollo.sh`-script is to make it easier to build ApolloOS ROM and different parts without rebuilding everything. It also makes it easier to build for both Vampire and UAE

**Comment by MBeijer** [comment]
@ronybeck
I've also added a feature to this build script so you can easily change the build-version by changing the data in the `version`-file in the repo root. :)

File: arch/m68k-amiga/hidd/amigavideo/amigavideo_hiddclass.c [hunk 1]
```diff
@@ -62,7 +62,7 @@ VOID AmigaVideoCl__Hidd_Gfx__NominalDimensions(OOP_Class *cl, OOP_Object *o, str
     if (msg->height)
         *(msg->height) = GfxBase->NormalDisplayRows;
     if (msg->depth)
-        *(msg->depth) = 1;
+        *(msg->depth) = 2;
```
Comment (MBeijer) [inline arch/m68k-amiga/hidd/amigavideo/amigavideo_hiddclass.c:65]: Fixes #31

**Comment by ronybeck** [comment]
There seems to be lots of exciting changes here.   Could you perhaps give an overview of what exactly is in this change is about?  It feels like there are multiple things rolled up into one merge request.  To help us understand what this merge is about it would very helpful if you could break it down into:
- What changes/bugfixes/new-features are contained within this merge request
- Why these changes should be applied together and not independently (i.e. separate merge requests).
- What the goal of this change is if it is obvious from the answers to the above
- Why we need a new build script (other than the old scripts are crap ;-) )
- What is this I2C stuff? :-)  It looks very interesting.

Some things to note are:
Rom space isn't infinite and adding new stuff to the rom really needs to be necessary and usable across all the vampire boards.  Submitting multiple changes in a single merge request is just a testing nightmare.  When something breaks, how can one test each of these changes individually to isolate which change causes the crash.

With regards to the mkapollo.sh script.........I know the existing ones are overly simple but this was intentional because the common complaint was that it is simply too hard to build AROS with the autotools (not all are familiar with Linux dev tools).  So those scripts are brain dead simple because that was what was needed: make building as simple as possible.   I think mkapollo.sh achieves a different goal, make it as flexible as possible maybe?   Wrapping autotools with an equally complex build script doesn't feel immediately advantageous.   Perhaps an explanation why mkapollo.sh is better with some examples would be nice.   Then we could ditch the old scripts in favour of this new one.

Is my request fair?

**Comment by MBeijer** [comment]
@ronybeck sure I can split it up. The new script was created by Alynna to make it easier to compile ApolloOS rom with different settings. 

I am aware of the ROM space limitations, and have been looking at which parts could be stripped out and tried different configurations to attain speed and still fit within the limitations, with various degrees of success.

The I2C stuff is to allow using I2C as a serial device and in extension use it for MIDI with camd.device.
I2C on Vampire -> Arduino -> Arduino translates to serial data at selected baud -> device.
Plan is to have two-way serial connectivity, aside from the serial debug output-only that is already existing on v4 at a fixed baud-rate.

As you can see, this PR is not for the master branch, but Alynna's branch, and is only a draft so to say.

This PR includes:
* I2C support on ROM-level, for usage from early boot (serial communications, debugging, etc)
* I2C support software, OS level
* Black and white -> Color fix in early boot (already committed in master)
* Compile error fixes
* Build system additions (config/make.tmpl)
* Easy changable version to display in Early Boot Selection by just changing the contents of a file called `version` in repo-root + build date ( rom/dosboot/menu.c )
* Early startup console shows which version of boot rom you have + build-date ( rom/dos/banner.c, rom/dos/boot.c )
* Fetch-script can download and knows how to unpack lha-archives
* VASM is included as a build tool, to enable building of ASM code for the ROM/distro
* fd2fsd is included as a build tool, to enable using of OS3.9 NDK for building ASM libraries
* lha unarchiving tool is included as a build tool, to enable extraction of LHA packages in the build process

This is very much work in progress, but the mkapollo.sh script:
* Makes sure dependencies are properly installed before building
* Decreases failure rate during build
* Enables easy versioning in early boot console
* Easier to build multiple version from same repo folder for both testing in UAE and on V4
* Makes it easy to list contents of ROM
* Bundles all build output in one folder and show you the output path
* Automatically combines ROM + EXT-ROM into a ApolloOS 1MB ROM for easy mapping on V4
* Installs required packages for building on Debian/Ubuntu
* Provides a way to easily get started for new developers

I suggest you test the mkapollo.sh script, it has a built-in help-screen with the available options for building

**Comment by MBeijer** [comment]
## **TODO move items to separate PR's:** 
### Checked = New PR made, ~~Striked~~ = Merged
- [ ] I2C support on ROM-level, for usage from early boot (serial communications, debugging, etc)
- [ ] I2C support software, OS level
- [x] ~~Black and white -> Color fix in early boot (already committed in master)~~
- [x] ~~Compile error fixes~~
- [x] ~~Build system additions (config/make.tmpl)~~
- [x] ~~Easy changable version to display in Early Boot Selection by just changing the contents of a file called `version` in repo-root + build date ( rom/dosboot/menu.c )~~ 
- [x] ~~Early startup console shows which version of boot rom you have + build-date ( rom/dos/banner.c, rom/dos/boot.c )~~
- [x] ~~Fetch-script can download and knows how to unpack lha-archives~~
- [x] ~~VASM is included as a build tool, to enable building of ASM code for the ROM/distro~~
- [ ] fd2fsd is included as a build tool, to enable using of OS3.9 NDK for building ASM libraries
- [x] ~~lha unarchiving tool is included as a build tool, to enable extraction of LHA packages in the build process~~
- [x] ~~`mkapollo.sh`-script~~

**Pull request closed without merging**